tests/test_peft_model.py

```diff
@@ -17,157 +17,132 @@
-                    **self.config_kwargs[i],
-                )
-                model = get_peft_model(model, config)
-
-                dummy_input = torch.LongTensor([[1, 1, 1]])
-                dummy_output = model.get_input_embeddings()(dummy_input)
-
-                self.assertTrue(dummy_output.requires_grad)
-
-    def test_save_pretrained(self):
-        r"""
-        A test to check if `save_pretrained` behaves as expected. This function should only save the state dict of the
-        adapter model and not the state dict of the base model. Hence inside each saved directory you should have:
-
-        - README.md (that contains an entry `base_model`)
-        - adapter_config.json
-        - adapter_model.bin
+    We use parametrized.expand for debugging purposes to test each model individually.
+    """
 
-        """
-        for model_id in self.checkpoints_to_test:
-            for i, config_cls in enumerate(self.config_classes):
-                model = AutoModelForCausalLM.from_pretrained(model_id)
-                config = config_cls(
-                    base_model_name_or_path=model_id,
-                    **self.config_kwargs[i],
+    @parameterized.expand(PeftTestConfigManager.get_grid_parameters(PEFT_MODELS_TO_TEST))
+    def test_attributes_parametrized(self, test_name, model_id, config_cls, config_kwargs):
+        self._test_model_attr(model_id, config_cls, config_kwargs)
+
+    def _test_model_attr(self, model_id, config_cls, config_kwargs):
+        model = AutoModelForCausalLM.from_pretrained(model_id)
+        config = config_cls(
+            base_model_name_or_path=model_id,
+            **config_kwargs,
+        )
+        model = get_peft_model(model, config)
+
+        self.assertTrue(hasattr(model, "save_pretrained"))
+        self.assertTrue(hasattr(model, "from_pretrained"))
+        self.assertTrue(hasattr(model, "push_to_hub"))
+
+    def _test_prepare_for_training(self, model_id, config_cls, config_kwargs):
+        model = AutoModelForCausalLM.from_pretrained(model_id).to(self.torch_device)
+        config = config_cls(
+            base_model_name_or_path=model_id,
+            **config_kwargs,
+        )
+        model = get_peft_model(model, config)
+
+        dummy_input = torch.LongTensor([[1, 1, 1]]).to(self.torch_device)
+        dummy_output = model.get_input_embeddings()(dummy_input)
+
+        self.assertTrue(not dummy_output.requires_grad)
+
+        # load with `prepare_model_for_int8_training`
+        model = AutoModelForCausalLM.from_pretrained(model_id).to(self.torch_device)
+        model = prepare_model_for_int8_training(model)
+
+        for param in model.parameters():
+            self.assertTrue(not param.requires_grad)
+
+        config = config_cls(
+            base_model_name_or_path=model_id,
+            **config_kwargs,
+        )
+        model = get_peft_model(model, config)
+        model = prepare_model_for_training(model)
```

Comment:
This has been removed, right?

Reply:
Added it in [4ff8380](https://github.com/huggingface/peft/pull/203/commits/4ff838003a79eecf11d840dec280bebc3177b7df)
I think we need an utility function to perform:
```python
if hasattr(model, "enable_input_require_grads"):
        model.enable_input_require_grads()
else:

        def make_inputs_require_grad(module, input, output):
            output.requires_grad_(True)

        model.get_input_embeddings().register_forward_hook(make_inputs_require_grad)
```
Happy to do it in a follow up PR